### PR TITLE
fix(dashboards): Move generated dashboards to bottom by default

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

We generate dashboards for users for all feature flags.

This can lead to annoyances when listing dashboards, as generated ones overwhelm your own.

This PR removes them from the list, and only allows editing these dashboards via the feature flag usage tab, where they were generated.

~~This PR makes it such that generated ones are at the bottom by default, unless:~~

1. You change the name (hence indicating interest in doing something more with the dashboard)
2. You're sorting by something relevant

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by liyiy — APPROVED** [review]
I'll let team PA be the final reviewer of this. This is definitely way better than having a bunch of unwanted dashboards polluting the table, though do people even click them from the dashboards page itself at all? Why not just filter them out for just the dashboards page completely? 🤔

**Comment by neilkakkar** [comment]
I wanted them around to let people edit things, but good point, I can link them directly to the dashboard to edit from the usage tab itself.

Will do that, and remove all mention from the list. 👍

File: frontend/src/models/dashboardsModel.tsx [hunk 1]
```diff
@@ -234,22 +235,40 @@ export const dashboardsModel = kea<dashboardsModelType>({
...
+                }
+                if (
+                    !firstName.startsWith(GENERATED_DASHBOARD_PREFIX) &&
+                    secondName.startsWith(GENERATED_DASHBOARD_PREFIX)
+                ) {
+                    return -1 // a should come before b
+                }
+
+                return firstName.localeCompare(secondName)
+            },
```
Comment (Twixes) [inline frontend/src/models/dashboardsModel.tsx:258]: We're gonna do a lot of these `startsWith` twice, which also makes this harder to read. Let's do some `firstNameIsGenerated` consts instead

File: frontend/src/models/dashboardsModel.tsx [hunk 2]
```diff
@@ -234,22 +235,40 @@ export const dashboardsModel = kea<dashboardsModelType>({
     },
 
     selectors: ({ selectors }) => ({
+        nameCompareFunction: [
+            () => [() => 1],
```
Comment (Twixes) [inline frontend/src/models/dashboardsModel.tsx:239]: Since this doesn't depend on the logic in any way, let's just pull it out as a function

**Review by Twixes — APPROVED** [review]
Makes a ton of sense, just a couple minor comments

File: frontend/src/models/dashboardsModel.tsx [hunk 3]
```diff
@@ -237,19 +238,17 @@ export const dashboardsModel = kea<dashboardsModelType>({
         nameSortedDashboards: [
             () => [selectors.rawDashboards],
             (rawDashboards) => {
-                return [...Object.values(rawDashboards)].sort((a, b) =>
-                    (a.name ?? 'Untitled').localeCompare(b.name ?? 'Untitled')
-                )
+                return [...Object.values(rawDashboards)]
+                    .filter((dashboard) => !(dashboard.name ?? 'Untitled').startsWith(GENERATED_DASHBOARD_PREFIX))
```
Comment (Twixes) [inline frontend/src/models/dashboardsModel.tsx:242]: Wait, if the ones starting with `GENERATED_DASHBOARD_PREFIX` are filtered out, it doesn't make sense to sort by that prefix, right?

**Review by Twixes — DISMISSED** [review]

File: frontend/src/models/dashboardsModel.tsx (hunk 3 above)
Comment (neilkakkar) [reply frontend/src/models/dashboardsModel.tsx:242]: Not here, but, this isn't the only place the comparison function is used, and I wanted to keep the function standard for sorting wherever you want to sort dashboards, so if, say, you're sorting raw dashboards somewhere, they're still at the end. Seemed less error prone.

Happy to remove generated prefix from the sorting function if you feel strongly about this

Comment (Twixes) [reply frontend/src/models/dashboardsModel.tsx:242]: Right, but `nameSortedDashboards` is the source for all our lists of dashboards, so if we filtered out the generated dashboards, they won't be displayed at the bottom anymore – they will be just hidden. From the description of the PR I thought that that's not the intention

Comment (neilkakkar) [reply frontend/src/models/dashboardsModel.tsx:242]: original intention yes, but based on Li's comment, am removing them completely from the list, and adding a link to the usage dashboard to get to the main db, will update description 😅

Comment (Twixes) [reply frontend/src/models/dashboardsModel.tsx:242]: Oh, I missed that, was still looking at the PR title 😅 In that case `nameCompareFunction` handling is unused in this PR, so would be better to revert that addition

Comment (neilkakkar) [reply frontend/src/models/dashboardsModel.tsx:242]: I'd argue that it at least unifies the current 3 places where we manually write the same function again, would you be fine with me removing the generated prefix from it but keeping the function, or do you want me to get rid of it completely?

Comment (Twixes) [reply frontend/src/models/dashboardsModel.tsx:242]: No, yeah, the function makes sense. Just the prefix handling that's not going to be used

**Comment by neilkakkar** [comment]
@Twixes 0.05 didn't help, so I bumped up to 0.08

**Pull request merged**